src/coreclr/nativeaot/Runtime/eventtrace.cpp

```diff
@@ -4829,13 +4829,13 @@ void ETW::LoaderLog::SendAssemblyEvent(Assembly *pAssembly, DWORD dwEventOptions
     BOOL bIsDynamicAssembly = pAssembly->IsDynamic();
     BOOL bIsCollectibleAssembly = pAssembly->IsCollectible();
     BOOL bIsDomainNeutral = pAssembly->IsDomainNeutral() ;
-    BOOL bHasNativeImage = pAssembly->GetManifestFile()->HasNativeImage();
+    BOOL bHasNativeImage = pAssembly->GetPEAssembly()->HasNativeImage();
 
     ULONGLONG ullAssemblyId = (ULONGLONG)pAssembly;
     ULONGLONG ullDomainId = (ULONGLONG)pAssembly->GetDomain();
     ULONGLONG ullBindingID = 0;
 #if (defined FEATURE_PREJIT) && (defined FEATURE_FUSION_DEPRECATE)
-    ullBindingID = pAssembly->GetManifestFile()->GetBindingID();
+    ullBindingID = pAssembly->GetPEAssembly()->GetBindingID();
```

Comment:
Hmm. I see some uses of `FEATURE_PREJIT`, I assumed it could be set in some obscure cases. Yes, this looks dead.

Reply:
It was removed some time last year - before NativeAOT made it over into the runtime repo, so I wouldn't be surprised if there are remnants in some of those sources.

Reply:
Most of eventtrace.cpp is ifdeffed out because `!FEATURE_REDHAWK`. I wouldn't spend time on making any changes to this file - this line is dead code.

We need to update the `FEATURE_REDHAWK` ifdefs in src\coreclr\vm\eventtrace.cpp and delete the NativeAOT fork.

Reply:
(The concept of assemblies, PE files, and binding doesn't exist in the NativeAOT runtime because all that happened ahead of time and the only thing left is code and data structures. We emulate assemblies for the reflection stack, but that happens in managed code.)

Reply:
I will just revert this then and let it be dealt with separately.